### PR TITLE
Update list.php

### PR DESCRIPTION
**Pull request by elfibro**
Added condition to include client code in the search when the global variable SOCIETE_ADD_REF_IN_LIST is set to 1.

- Modified the SQL query to include 's.code_client' in the search conditions.
- Added check for the global variable SOCIETE_ADD_REF_IN_LIST.
- If SOCIETE_ADD_REF_IN_LIST is 1, 's.code_client' is included in the search along with 's.nom' and 's.name_alias'.
- Ensured that 's.code_client' is searched separately only when SOCIETE_ADD_REF_IN_LIST is 1 and 'search_societe' is not defined.

# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE", "NEW", "PERF" or "QUAL" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*

# FIX|Fix #[*issue_number Short description*]
[*Long description*]

# CLOSE|Close #[*issue_number Short description*]
[*Long description*]

# NEW|New [*Short description*]
[*Long description*]

# PERF|Perf #[*issue_number Short description*]
[*Long description*]

# QUAL|Qual #[*issue_number Short description*]
[*Long description*]

### CODE REVIEW

File: htdocs/comm/propal/list.php [hunk 1]
```diff
@@ -635,16 +635,29 @@
 if ($search_availability) {
 	$sql .= " AND p.fk_availability IN (".$db->sanitize($db->escape($search_availability)).')';
 }
+$societe_add_ref_in_list = (!empty($conf->global->SOCIETE_ADD_REF_IN_LIST) ? $conf->global->SOCIETE_ADD_REF_IN_LIST : 0); // Récupérer la valeur de SOCIETE_ADD_REF_IN_LIST
```
Comment (eldy) [inline htdocs/comm/propal/list.php:638]: Can you replace conf->global->xxx with getDolGlobalString('xxx
Old syntax is no more allowed.

**Review by eldy — CHANGES_REQUESTED** [review]

File: htdocs/comm/propal/list.php (hunk 1 above)
Comment (elfibro) [reply htdocs/comm/propal/list.php]: It's ok now

File: htdocs/comm/propal/list.php [hunk 2]
```diff
@@ -635,16 +635,29 @@
 if ($search_availability) {
 	$sql .= " AND p.fk_availability IN (".$db->sanitize($db->escape($search_availability)).')';
 }
+$societe_add_ref_in_list = (!empty(getDolGlobalString('SOCIETE_ADD_REF_IN_LIST')) ? getDolGlobalString('SOCIETE_ADD_REF_IN_LIST') : 0); // Récupérer la valeur de SOCIETE_ADD_REF_IN_LIST
```
Comment (frederic34) [inline htdocs/comm/propal/list.php:638]: `$societe_add_ref_in_list = getDolGlobalInt('SOCIETE_ADD_REF_IN_LIST');` is more simple and pass the tests

Comment (elfibro) [reply htdocs/comm/propal/list.php]: It's ok now

**Pull request merged**